src/main/java/org/lsc/utils/ScriptingEvaluator.java

```diff
@@ -115,13 +116,33 @@ public static void contribute(String implementationName,
 
 	private ScriptableEvaluator identifyScriptingEngine(String expression) throws LscServiceException {
 		String[] parts = expression.split(":");
-		if (parts != null && parts.length > 0 && parts[0].length() < 10
-				&& instancesTypeCache.containsKey(parts[0])) {
-			return instancesTypeCache.get(parts[0]);
+		String match = matchJScriptEvaluator(parts[0]);
+		if (!match.isEmpty()) {
+			return instancesTypeCache.get(match);
 		}
 		return defaultImplementation.orElseThrow(() -> new LscServiceException("Missing Script evaluator"));
 	}
 
+	/**
+	 * Matches the prefix specifying the jscript evaluator.
+	 *
+	 * @param jscript the prefix
+	 * @return the matched jscript evaluator
+	 */
+	public String matchJScriptEvaluator(String jscript) {
+		for (String jscriptEngine : instancesTypeCache.keySet()) {
+			Pattern pattern = Pattern.compile("((\\n.*)+)" + jscriptEngine);
```

Comment:
As far as I know it's recommended to compile patterns only once, as it can be a heavy computation.

Reply:
Created a method to pre-compile all regex pattern match combinations.